explorer/syntax/parser.ypp

```diff
@@ -1059,14 +1064,28 @@ type_params:
 | tuple_pattern
     { $$ = $1; }
 ;
+class_declaration_prefix:
```

Comment:
Seems to get the job done (tests passing), but I may be missing some subtleties of the parser.

Reply:
No, this approach seems reasonable to me. I'm pretty sure these are mutually exclusive.